src/gpu/nvidia/sycl_cuda_compat.hpp

```diff
@@ -42,11 +42,23 @@ void host_task(::sycl::handler &cgh, const T &task) {
 
 template <typename native_object_t, typename sycl_object_t>
 native_object_t get_native(const sycl_object_t &sycl_object) {
+    static_assert(!std::is_same_v<native_object_t, CUcontext>,
+            "Use compat::get_native_context() to obtain the native cuda "
+            "context");
     auto handle
             = ::sycl::get_native<::sycl::backend::ext_oneapi_cuda>(sycl_object);
     return reinterpret_cast<native_object_t>(handle);
 }
 
+inline CUcontext get_native_context(const ::sycl::device &sycl_device) {
+    CUdevice cu_device = get_native<CUdevice>(sycl_device);
+    CUcontext cu_ctx;
+    const CUresult cu_result = cuDevicePrimaryCtxRetain(&cu_ctx, cu_device);
```

Comment:
> Retains the primary context on the device. Once the user successfully retains the primary context, the primary context will be active and available to the user until the user releases it with [cuDevicePrimaryCtxRelease()](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__PRIMARY__CTX.html#group__CUDA__PRIMARY__CTX_1gf2a8bc16f8df0c88031f6a1ba3d6e8ad) or resets it with [cuDevicePrimaryCtxReset()](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__PRIMARY__CTX.html#group__CUDA__PRIMARY__CTX_1g5d38802e8600340283958a117466ce12).

It looks like the retained context should be released at some point but I don't see we do that. Can you clarify this?